src/libcore/char.rs

```diff
@@ -269,10 +269,10 @@ pub trait CharExt {
     fn len_utf8(self) -> usize;
     #[stable(feature = "core", since = "1.6.0")]
     fn len_utf16(self) -> usize;
-    #[stable(feature = "core", since = "1.6.0")]
-    fn encode_utf8(self, dst: &mut [u8]) -> Option<usize>;
-    #[stable(feature = "core", since = "1.6.0")]
-    fn encode_utf16(self, dst: &mut [u16]) -> Option<usize>;
+    #[unstable(feature = "unicode", issue = "27784")]
```

Comment:
Oh right yes thanks for the reminder! I meant to discuss this more at length on the PR comments but forgot.

So it looks like `char::encode_utf8` is unstable (as expected), but we accidentally stabilized `CharExt::encode_utf8` in libcore. This appears to be accidental (from what I can tell), so it seems that we're within our rights to switch back to unstable here (with libstd being the "source of truth")

That being said I wouldn't mind doing a crater run to see if it affects any crates.